t/test_cleanupSpies

```diff
@@ -0,0 +1,48 @@
+#!/bin/sh
+
+readonly TEST_DIR="$(dirname "$0")"
+
+# shellcheck source=/dev/null
+. "$TEST_DIR/../shpy"
+# shellcheck source=/dev/null
+. "$TEST_DIR/../shpy-shunit2"
+
+testRemovesDirectory() {
+  createSpy foo
+
+  # shellcheck disable=SC2154
+  assertNotNull "${_shpy_spies_dir}"
```

Comment:
Private variables (those starting with an underscore, which also isn't documented anywhere…) shouldn't be used directly in testing. Instead, you could try:

- Stubbing the `mktmp` call to return a known directory and verifying that directory is removed
- Do a pattern match for a directory named `shpy.[0-9]+`, keeping in mind that `zsh` doesn't glob like other shells
- Stub all the i/o functions to verify a call for a tmp directory was made and a call to rm it was made

It's likely that `$_shpy_spies_dir` reflects the directory on disk, but it's best to check for the directory itself because that's what the end user would see 🔍